### ISSUE TITLE
Screenreader activation status is not captured when application starts after the client

### ISSUE THREAD

**Issue by deepak1556** [microsoft/vscode#172690]
Extracted from https://github.com/microsoft/vscode/issues/167627#issuecomment-1407000299

Steps to Reproduce:

1. Enable Voice Over
2. Start VSCode Insiders
3. Application has correctly entered a11y mode, since Voice Over is able to read the contents
4. Status bar does not reflect the a11y mode

Version: 1.75.0-insider (Universal)
Commit: da0309f2514ac22ac82dcd080cfc764fe2a1f434
Date: 2023-01-27T14:19:56.997Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 22.2.0
Sandboxed: Yes

**Comment by deepak1556** [comment]
The problem here is that we register the listener to receive a11y event from the runtime [here](https://github.com/microsoft/vscode/blob/b5aa3e0a3d3cdac2c44c022182ca2f238949a444/src/vs/code/electron-main/app.ts#L326-L329) but in this particular scenario the event is emitted before the [ready event](https://github.com/microsoft/vscode/blob/b5aa3e0a3d3cdac2c44c022182ca2f238949a444/src/main.js#L116) hence we miss the signal.

@bpasero need to pick your mind here, I would like to move this event handler to `main.js` file before the ready event. What is the best way to update `editor.accessibilitySupport` configuration at this stage ?

**Comment by bpasero** [comment]
I think a hack like this:

https://github.com/microsoft/vscode/blob/b69204b36a2ca3eb4a0ad9cfaa6dd101ca96d4ae/src/main.js#L465

https://github.com/microsoft/vscode/blob/b69204b36a2ca3eb4a0ad9cfaa6dd101ca96d4ae/src/vs/code/electron-main/app.ts#L1128

Because there is no way to talk to windows so early. 

I think we can also set this early when creating the window in case the event is sent even before we open the first window:

https://github.com/microsoft/vscode/blob/b69204b36a2ca3eb4a0ad9cfaa6dd101ca96d4ae/src/vs/platform/windows/electron-main/windowImpl.ts#L1086

**Comment by deepak1556** [comment]
Thanks @bpasero , moving the a11y support configuration to https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/platform/windows/electron-main/windowImpl.ts#L1014-L1059 can cover both the reload as well as early trigger scenario since they are calculated for each window load operation.  

However, even with the above change both https://github.com/microsoft/vscode/issues/172688 and this issue suffer from the same root cause, i-e [NativeAccessibilityService](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/workbench/services/accessibility/electron-sandbox/accessibilityService.ts#L30) is initialized earlier than [AccessibilityStatus](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/workbench/browser/parts/editor/accessibilityStatus.ts#L19) causing the observer to miss the initial [_onDidChangeScreenReaderOptimized](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/platform/accessibility/browser/accessibilityService.ts#L112) event.

**Comment by meganrogge** [comment]
@deepak1556 did you make the change that you describe above or shall I?

and @bpasero is there any way to change this? 

> [NativeAccessibilityService](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/workbench/services/accessibility/electron-sandbox/accessibilityService.ts#L30) is initialized earlier than [AccessibilityStatus](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/workbench/browser/parts/editor/accessibilityStatus.ts#L19) causing the observer to miss the initial [_onDidChangeScreenReaderOptimized](https://github.com/microsoft/vscode/blob/3ec25a9fbb92d89e388c151ed56b4cca643b1cc7/src/vs/platform/accessibility/browser/accessibilityService.ts#L112) event.

**Comment by deepak1556** [comment]
@meganrogge I haven't made any changes, please feel free to make the configuration changes as well as part of the fix.

**Comment by bpasero** [comment]
Maybe I am missing something, but why can the `NativeAccessibilityService` not cache the state of accessibility and then allow `AccessibilityStatus` to ask for it instead of relying on an event that has already fired?

**Comment by deepak1556** [comment]
> Maybe I am missing something, but why can the NativeAccessibilityService not cache the state of accessibility and then allow AccessibilityStatus to ask for it instead of relying on an event that has already fired?

I think this would work. The scenario described here as well as in https://github.com/microsoft/vscode/issues/172688 are when the event has already fired when initializing the workbench, in those cases querying the status from the `AccessibilityService` after construction should do the trick. For other normal execution cases, the event will be a reliable indicator.
